documentation/src/main/antora/modules/ROOT/pages/component-checkpoint.adoc

```diff
@@ -0,0 +1,279 @@
+
+
+Sample configuration class with checkpointing annotations:
+[source,java]
+----
+    ...
+    @Checkpoint
+    @Version(value = 2, migrationHandler = CheckpointMigrationHandler.class)
+    public static class CheckPointInputConfig implements Serializable {
+
+        public enum Strategy {
+            BY_ID,
+            BY_DATE
+        }
+
+        @Option
+        @DefaultValue("BY_ID")
+        private Strategy strategy = Strategy.BY_ID;
+
+        @Option
+        private String startDate;
+
+        @Option
+        private int sinceId;
+
+        @Option
+        private String status;
+    }
+
+    @Data
+    @GridLayout(value = { @GridLayout.Row("user"), @GridLayout.Row("pass")})
+    @GridLayout(names = GridLayout.FormType.CHECKPOINT, value = { @GridLayout.Row("checkPointInputConfig") })
+    public static class InputConfig {
+
+        @Option
+        private String user;
+
+        @Option
+        private String pass;
+
+        @Option
+        private CheckPointInputConfig checkpoint = new CheckPointInputConfig();
+    }
+----
+
+Calling the `getCheckpoint()` returns `CheckpointState` class, which is a simple POJO class that holds the checkpoint data and version. This class has to be serializable to allow the runtime to save and restore the checkpoint object. For that, it provides a helper method `toJson()` to serialize the checkpoint state object.
+
+Here's a simple example of a `CheckpointState` class serialized to JSON:
+[source,json]
+----
+{
+  "$checkpoint": { <1>
+    "lastId": 95,
+    "lastUpdate": "2023-04-04",
+    "status": "running",
+    "strategy": "BY_ID",
+    "__version": 2 <2>
+  }
+}
+----
+Notice the following which are important (automatically done using `toJson()`):
+
+<1> Configuration and state of the checkpoint should be a member of a `$checkpoint` object.
+<2> Version of the checkpoint configuration is stored for eventual migration.
+
+The checkpointing configuration class is defined as a nested class within the component configuration class. The runtime has to provide the checkpointing configuration to the component when it is started.
+[source,java]
+----
+  ...
+  configuration.put("$checkpoint.strategy", "BY_ID");
+  configuration.put("$checkpoint.sinceId", "5");
+  configuration.put("$checkpoint.status", "none");
+  //
+  final Mapper mapper = mgr.findMapper("checkpoint", "list-input", 1, configuration).get();
+  ...
+----
+
+Important points to remember when providing the configuration to the component, you don't need to respect the internal configuration path of checkpoint configuration class in your connector. The runtime will automatically map the configuration to the checkpoint configuration class when you prefix your checkpoint state with `$checkpoint`.
```

Comment:
Ok, please give a full example, it is important I think.

Reply:
enriched configuration above.